src/main/kotlin/com/autonomousapps/tasks/ResolveExternalDependenciesTask.kt

```diff
@@ -51,17 +58,30 @@ abstract class ResolveExternalDependenciesTask : DefaultTask() {
     runtimeClasspath: Configuration,
     jarAttr: String
   ) {
-    this.compileClasspath = compileClasspath
-    this.runtimeClasspath = runtimeClasspath
+    compileClasspathResult.set(compileClasspath.incoming.resolutionResult.rootComponent)
```

Comment:
This is the main logic change: move to `Property` API with types that are configuration cache compatible.
Set the value in the `configureTask` method in a way that makes sure the evaluation is fully lazy.